lib/vanagon/common/pathname.rb

```diff
@@ -28,15 +28,18 @@ class Pathname
       # @param [String, Integer] group the groupname or GID to use when this file is archived
       # @param [Boolean] config mark this file as a configuration file, stored as private state
       #   and exposed through the {#configfile?} method.
+      # @param [string] id the id to pass to wix objects during directory generation
       # @return [Vanagon::Common::Pathname] Returns a new Pathname instance.
-      def initialize(path, mode: nil, owner: nil, group: nil, config: false)
+      def initialize(path, mode: nil, owner: nil, group: nil, config: false, id: nil)
```

Comment:
I think you're going to want to update https://github.com/puppetlabs/vanagon/blob/f7a2b4ccdf2edd5a28ef49a469f662a6252b2a60/lib/vanagon/project/dsl.rb#L146 as well to pass in `id`?